apm-agent-plugins/apm-urlconnection-plugin/src/test/java/co/elastic/apm/agent/urlconnection/HttpUrlConnectionInstrumentationTest.java

```diff
@@ -121,4 +121,27 @@ public void testFakeReuse() throws Exception {
         verifyHttpSpan("/");
     }
 
+    @Test
+    public void testGetResponseCodeWithUnhandledException() {
+        try {
+            final HttpURLConnection urlConnection = (HttpURLConnection) new URL("http://unknown").openConnection();
+            urlConnection.getResponseCode();
+            urlConnection.disconnect();
+        } catch (Exception e) {
+            // intentionally ignored
+        }
+
+        assertThat(reporter.getErrors()).hasSize(1);
+    }
+
+    @Test
+    public void testGetResponseWithHandledException() throws Exception {
+        final HttpURLConnection urlConnection = (HttpURLConnection) new URL(getBaseUrl() + "/non-existent").openConnection();
+        urlConnection.getResponseCode();
+        urlConnection.disconnect();
+
+        verifyHttpSpan("localhost", "/non-existent", 404);
+        assertThat(reporter.getErrors()).isEmpty();
+    }
```

Comment:
Please add a test for the same URL only using `getInputStream()` (or even better- use `performGet()`) instead of `getResponseCode()` and verifying the expected behavior - `assertThat(reporter.getErrors()).hasSize(1)`

Reply:
added, but kind like redundant to `co.elastic.apm.agent.httpclient.AbstractHttpClientInstrumentationTest#testNonExistingHttpCall` except for the additional assert. So i am not sure if changing the mentioned test, would also be an option.

Reply:
The additional assertion (for an error being captured) is exactly what makes it non-redundant, it is what verifies this codes does what we intend it does.
And changing the mentioned test is not preferable as this is a very specific behavior for `HttpUrlConnection` instrumentation and only when using `getInputStream`. Other clients do not report errors in such 404 requests.

Reply:
> And changing the mentioned test is not preferable as this is a very specific behavior for `HttpUrlConnection` instrumentation and only when using `getInputStream`. Other clients do not report errors in such 404 requests.

True, that makes sense then.